### ISSUE TITLE
Lifecycle of Sidecar containers

### ISSUE THREAD

**Issue by ImpSy** [GoogleCloudPlatform/spark-on-k8s-operator#800]
Hello,

I'm currently having an issue using Sidecars:
I've created a Sidecar to tail `spark.eventLog` but the container never stop even when the driver is finished

My configuration as follow:
```
sparkConf:
  spark.eventLog.enabled: "true"
driver:
  volumeMounts:
  - name: event-log
    mountPath: "/tmp/spark-events"
  sidecars:
  - name: event-log-collector
    image: busybox
    args:
    - "/bin/sh"
    - "-c"
    - "sleep 30 && tail -f /tmp/spark-events/*.inprogress"
    volumeMounts:
    - name: event-log
      mountPath: "/tmp/spark-events"
volumes:
- name: event-log
  emptyDir: {}
```

Is this a bug or just not implemented ?
If that's the case, do you know any trick to stop the sidecar ?

Thanks in advance for your response 😃

**Comment by liyinan926** [comment]
Lifecycle support for sidecars has not been implemented yet. We add sidecars as normal containers without adding the following:

```yaml
lifecycle:
  type: Sidecar
```

**Comment by ImpSy** [comment]
Ok, thanks for the quick response 👌 

Another question:
Since the Sidecar Lifecycle will only available for kubernetes 1.18 and above, will it ever be possible to access Lifecycle Handlers from the driver container (espacially `preStop`) ?

That way we could use this [trick](https://github.com/kubernetes/kubernetes/issues/25908#issuecomment-365924958) to handle the lifecycle of Sidecars prior to 1.18

**Comment by liyinan926** [comment]
No, we don't currently support `postStart` and `preStop` lifecycle hooks for the driver/executor containers.
